File: leeaa/66.py
class Solution:
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        carry = 1
        for i in range(len(digits)-1, -1, -1):
            digits[i] += carry
            carry = int(digits[i]/10)
            digits[i] %= 10

        if carry:
            digits = [1] + digits

        return digits

A = Solution()
arr = [1,2,3,4]
print(A.plusOne(arr))

File: rein/66.js
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
  let carry = 1;
  digits = digits.slice();

  for (let i = digits.length - 1; i >= 0; --i) {
    digits[i] += carry;
    if (digits[i] >= 10) {
      digits[i] -= 10;
      carry = 1;
    } else {
      carry = 0;
    }
  }

  if (carry == 1) {
    digits.unshift(1);
  }

  return digits;
};


const arrayEqual = (a, b) => a.every((v, i) => v == b[i]);

const assert = require('assert');
console.log(plusOne([1,2,3]) , [1,2,4]);
console.log(plusOne([4,3,2,1]) , [4,3,2,2]);
console.log(plusOne([1,9,9,9]) , [2,0,0,0]);
console.log(plusOne([9,9,9]) , [1,0,0,0]);
